cpp/arcticdb/async/async_store.hpp

```diff
@@ -222,28 +222,47 @@ void iterate_type(
     library_->iterate_type(type, func, prefix);
 }
 
-folly::Future<storage::ObjectSizes> get_object_sizes(KeyType type, const std::string& prefix) override {
+folly::Future<folly::Unit> visit_object_sizes(
+    KeyType type, const std::optional<StreamId>& stream_id_opt, storage::ObjectSizesVisitor visitor) override {
+    std::string prefix;
+    if (stream_id_opt) {
+        auto stream_id = *stream_id_opt;
```

Comment:
I think we can do
```suggestion
        const auto& stream_id = *stream_id_opt;
```
to avoid copying the stream id